### PR TITLE
allow projects to change jvm version of compiled artifacts

### PR DESCRIPTION
**Pull request by SamRoberts**
Bureaucracy needs to run one of it's artifacts (oracle plugin) on jdk 6.

### CODE REVIEW

**Comment by SamRoberts** [comment]
I have tested this on bureaucracy.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala [hunk 1]
```diff
@@ -17,4 +17,5 @@ package au.com.cba.omnia.uniform.core.scala
 object Scala {
   val version = "2.10.5"
   val binaryVersion = "2.10"
+  val jvmVersion = "1.7"
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala]: Line up `=` :smile:.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala [hunk 2]
```diff
@@ -32,12 +35,12 @@ object ScalaSettings extends Plugin {
...
         "-feature",
         "-language:_",
-        "-target:jvm-1.7"
+        s"-target:jvm-$jvmVersion"
       ),
       javacOptions ++= Seq(
         "-Xlint:unchecked",
-        "-source", "1.7",
-        "-target", "1.7"
+        "-source", Scala.jvmVersion,
```
Comment (stephanh) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala]: What is the difference between `jvmVersion` and `Scala.jvmVersion`?

**Comment by stephanh** [comment]
My preference would be that we make the jvm version a setting instead of a parameter on `project` since it is going to be only really used by bureaucracy. However, I am happy for the PR to go in as is.

:+1:.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala (hunk 2 above)
Comment (SamRoberts) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/setting/ScalaSettings.scala]: One is a constant in the `Scala` object, the other is the parameter to this method. I should probably name the parameter differently! Will do ...

**Comment by SamRoberts** [comment]
Hrmm, using a separate setting is not a bad point. It would be the most canonical way to do it in sbt (unfortunately!). I will make the change.

File: uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala (hunk 1 above)
Comment (SamRoberts) [inline uniform-core/src/main/scala/au/com/cba/omnia/uniform/core/scala/Scala.scala]: Will do

**Comment by SamRoberts** [comment]
I am closing this pull request for now, as there's not much point testing it on bureaucracy until we'll ported bureaucracy to scala 2.11. I will test this once bureaucracy upgrades.

**Pull request closed without merging**